### PR TITLE
Handle core.commentChar in commit and rebase editors

### PR DESCRIPTION
**Pull request by lechat**
add config variable to hide commit help - disable_editor_help

I have the same problem as #320, I have to change `core.commitChar` to ";" in git. This PR fixes problem in commit and rebase editors.

If it seems that adding common function to `neogit.lib.git` is not appropriate, happy to change that to some other place.

### CODE REVIEW

File: lua/neogit/buffers/editor/init.lua [hunk 1]
```diff
@@ -77,26 +77,27 @@ function M:open(kind)
...
-      end)
+      if not config.values.disable_editor_help then
+        local comment_char = git.config.get_comment_char()
+        -- stylua: ignore
+        local help_lines = {
+          string.format("%s", comment_char),
+          string.format("%s Commands:", comment_char),
+          string.format("%s   %s Close", comment_char, pad_mapping("Close")),
+          string.format("%s   %s Submit", comment_char, pad_mapping("Submit")),
+          string.format("%s   %s Abort", comment_char, pad_mapping("Abort")),
```
Comment (CKolkey) [inline lua/neogit/buffers/editor/init.lua:88]: I think the `Previous`, `Next`, and `Reset` commands got dropped on accident.

File: lua/neogit/lib/git/config.lua [hunk 2]
```diff
@@ -147,4 +147,16 @@ function M.unset(key)
...
+  local char = M.get("core.commentChar").value
+  if char == "" then
+    char = M.get_global("core.commentChar").value
+    if char == "" then
+      char = "#"
+    end
+  end
+
+  return char
+end
```
Comment (CKolkey) [inline lua/neogit/lib/git/config.lua:160]: I don't really wanna have bespoke value getters in this library, but we can simplify how it's written in the callers to obviate the need for a function:

```lua
local comment_char = git.config.get("core.commentChar"):read() or git.config.get_global("core.commentChar"):read() or "#"
```

File: lua/neogit/config.lua [hunk 3]
```diff
@@ -135,6 +135,7 @@ end
 ---@field fetch_after_checkout? boolean Perform a fetch if the newly checked out branch has an upstream or pushRemote set
 ---@field telescope_sorter? function The sorter telescope will use
 ---@field disable_insert_on_commit? boolean|"auto" Disable automatically entering insert mode in commit dialogues
+---@field disable_editor_help? boolean|"auto" Disable displaying commit/rebase editor help
```
Comment (CKolkey) [inline lua/neogit/config.lua:138]: I don't think I want to expose this - at least not now.

**Review by CKolkey — CHANGES_REQUESTED** [review]
Ok! We can do something about this. Left some notes

File: lua/neogit/buffers/editor/init.lua (hunk 1 above)
Comment (lechat) [reply lua/neogit/buffers/editor/init.lua:88]: Not dropped by me. I just changed what was there.

File: lua/neogit/lib/git/config.lua (hunk 2 above)
Comment (lechat) [reply lua/neogit/lib/git/config.lua:160]: That's exactly what I was asking about. Will change code in a moment.

File: lua/neogit/config.lua (hunk 3 above)
Comment (lechat) [reply lua/neogit/config.lua:138]: Will remove this line.

File: lua/neogit/buffers/editor/init.lua (hunk 1 above)
Comment (lechat) [reply lua/neogit/buffers/editor/init.lua:88]: Ah, there were more changes in editor. Will  fix that as well.

Comment (CKolkey) [reply lua/neogit/buffers/editor/init.lua:88]: All good, I merged it very recently, probably some time between you forking and opening a PR :)

**Comment by CKolkey** [comment]
Thanks a ton for the contribution :)

**Pull request merged**